src/modules/roc_core/target_musl/roc_core/backtrace.cpp

```diff
@@ -6,15 +6,218 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+#ifndef LIBUNWIND_HH
+#define LIBUNWIND_HH
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+#endif
+#include <cxxabi.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
 #include "roc_core/backtrace.h"
 
+/* Function to convert unsigned int to decimal/hex string.
+ * This function is signal safe.
+ * Parameters : 
+ * number = the number to be converted.
+ * base = 10 for decimal to char* and 16 for hex to char* conversion.
+ * output_string = the decimal/hex string
+ */
+void uint_to_string(unsigned int number, int base, char *output_string)
+{
+	int i = 12;
```

Comment:
When the function is called with a 10-byte buffer:

```
		char index_buffer[10];
		uint_to_string(index, 10, index_buffer);
```

we're smashing the stack here (making a write of of the index_buffer bounds):

```
output_string[i] = "0123456789abcdef"[number % base];
```

Why you set it to 12 here? For 10 base, 4-byte unsigned integer can contain at most 10 digits, so the buffer size should be 11 (10 digits and `\0`) and `i` should be initialized to 10, isn't it?

I suggest to fix the function as follows:

* add `MaxDigits` constant and use it both inside the function and when calling it
* change `unsigned int` to `uint32_t` so that `MaxDigits` will be platform-independent
* also rename the function to `uint32_to_string`

```
enum { MaxDigits = 10 };

void uint32_to_string(uint32_t number, int base, char *output_string)
{
	int i = MaxDigits;
	int j = 0;
	do {
		output_string[i] = "0123456789abcdef"[number % base];
		i--;
		number = number/base;
	} while(number > 0);

	while(++i < MaxDigits + 1)
		output_string[j++] = output_string[i];
	output_string[j] = 0;
}

...

char index_buffer[MaxDigits+1];
uint32_to_string(somenum, 10, index_buffer);
```

Reply:
Yes, I overlooked that `i=12` part. Will be fixed.